react/week4/github-userSearch/src/components/UserData.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from "react";
+import UserSearch from "./Form";
+import UserList from "./UserList";
+
+function UserData() {
+	const url = `https://api.github.com/search/users?`;
+	const [search, setSearch] = useState("");
+	const [users, setUsers] = useState([]);
+	const [loading, setLoading] = useState(false);
+
+	const fetchUser = async () => {
+		let searchUrl = `${url}q=${search}`;
+		if (search) {
+			setLoading(true);
+			await fetch(searchUrl)
+				.then(data => data.json())
+				.then(data => {
+					setUsers(data.items);
+				});
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		fetchUser();
+	}, [search]);
+
+	const handleSearch = input => {
+		setSearch(input);
+	};
+	return (
+		<div>
+			<UserSearch handleSearch={handleSearch} />

```

Comment:
We don't need this function `handleSearch()`, we could use directly the state function
```suggestion
			<UserSearch handleSearch={setSearch} />
```

Reply:
Why was this resolved? 🤔  The problem is this there.